plugins/modules/volume.py

```diff
@@ -1658,10 +1658,15 @@ def get_diff_after(self, volume_params, volume_details, fetched_params, before_d
                         "source_timestamp": None
                     },
                     "protection_policy": {
-                        "id": "",
+                        "id": fetched_params['protection_policy_id'],
```

Comment:
please use .get() function whenever getting value from dict, which help in avoiding keyError when key is not in dict.

Reply:
This key will always be present, if the policy ios no provided, it will be None.